streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -131,6 +139,16 @@ public boolean hasPersistentGlobalStore() {
         return false;
     }
 
+    public void updateSourceTopics(final Map<String, List<String>> nodeToSourceTopics) {
```

Comment:
For this update function, should we also look at the `sourceNodesByName` and make sure their keysets are matching, otherwise we would throw illegal state here? I think this is legitimate insurance before @cadonna starts the topology revolution work.

Reply:
good call